lighthouse-cli/test/smokehouse/report-assert.js

```diff
@@ -199,14 +199,25 @@ function pruneExpectations(localConsole, lhr, expected) {
   }
 
   const cloned = cloneDeep(expected);
+
+  if (!isSync && expected.networkRequests) {
```

Comment:
a comment here on the reason for conditionally deleting will also probably be handy at some point, since it's not as self-explanatory as `_minChromiumMilestone`